src/Domain/Event/DomainEventFactory.ts

```diff
@@ -25,4 +26,18 @@ export class DomainEventFactory implements DomainEventFactoryInterface {
       },
     }
   }
+
+  createUserRoleChangedEvent(userUuid: string, email: string, fromRole: RoleName, toRole: RoleName): UserRoleChangedEvent {
+    return {
+      type: 'USER_ROLE_CHANGED',
+      createdAt: dayjs.utc().toDate(),
+      payload: {
+        userUuid,
+        email,
+        fromRole,
+        toRole,
+        timestamp: dayjs.utc().valueOf(),
```

Comment:
this should use the `TimerInterface` implementation so something like in here: https://github.com/standardnotes/auth/blob/5b6bfe5769fa69e30088a5cb0c2b3c6f2a45ca82/src/Domain/Setting/SettingFactory.ts#L19

Reply:
Awesome, will do! 🙂